### PR TITLE
OSC: Move HTML escaping from DB to frontend

### PR DESCRIPTION
**Pull request by lscharmer**
This PR changes that messages aren't escaped when writing them to the database but are saved unescaped in the database and are escaped when outputting them in HTML instead.

### CODE REVIEW

File: components/ILIAS/OnScreenChat/resources/onscreenchat.js [hunk 1]
```diff
@@ -1022,16 +1038,23 @@
 
 			let messageDate = new Date();
 			messageDate.setTime(messageObject.timestamp);
+			const n = 'm' + new Date().getTime();
```
Comment (mjansenDatabay) [inline components/ILIAS/OnScreenChat/resources/onscreenchat.js]: We should find another variable name for `n`.

File: components/ILIAS/OnScreenChat/classes/Setup/UpdateSteps.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    public function step_1(): void
+    {
+        $replace = [
+            '&lt;' => '<',
+            '&gt;' => '>',
+            '&amp;' => '&',
+            '&quot;' => '"',
+        ];
+
+        $s = 'message';
```
Comment (mjansenDatabay) [inline components/ILIAS/OnScreenChat/classes/Setup/UpdateSteps.php]: Can we find a better name fror `$s`?

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]
Hi @lscharmer ,

please have a look at `components/ILIAS/OnScreenChat/classes/Provider/OnScreenChatProvider.php` and check whether or not we have to introduce proper output escpaing for the main bar entry when presenting conversation messages in the slate.

Best regards,
Michael

File: components/ILIAS/OnScreenChat/resources/onscreenchat.js (hunk 1 above)
Comment (lscharmer) [reply components/ILIAS/OnScreenChat/resources/onscreenchat.js]: Maybe `m` ;)

Comment (mjansenDatabay) [reply components/ILIAS/OnScreenChat/resources/onscreenchat.js]: :facepalm:

File: components/ILIAS/Chatroom/chat/SocketTasks/ConversationMessage.js [hunk 3]
```diff
@@ -1,3 +1,19 @@
...
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
 var Container  = require('../AppContainer');
 var HTMLEscape = require('../Helper/HTMLEscape');
```
Comment (mjansenDatabay) [inline components/ILIAS/Chatroom/chat/SocketTasks/ConversationMessage.js]: Can we remove this if `HTMLEscape.escape` is not used anymore?

**Comment by lscharmer** [comment]
> Hi @lscharmer ,
> 
> please have a look at `components/ILIAS/OnScreenChat/classes/Provider/OnScreenChatProvider.php` and check whether or not we have to introduce proper output escpaing for the main bar entry when presenting conversation messages in the slate.
> 
> Best regards, Michael

Hi @mjansenDatabay,

messages shown in the slate are rendered with `$DIC->ui()->factory()->item()->shy()` and the `withDescription` method is used for displaying the chat message. This method html escapes the given description already. So currently messages are displayed double escaped in the menu slate. With this PR they will be only escpaed once.

Best regards
@lscharmer

**Pull request merged**